File: endModule/fe/src/app/savingBook/module/saving-book.ts
// tslint:disable-next-line:no-empty-interface
import {Customer} from './customer';

export interface SavingBook {
    id?: number;
    openingDay?: Date ;
    startDay?: Date ;
    period?: number;
    deposit?: number;
    reference?: string;
    interestRate?: number;
    customer?: Customer;
}

File: endModule/backEnd/src/main/java/com/example/backend/model/SavingBook.java
package com.example.backend.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Getter
@Setter
public class SavingBook {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date openingDay;
    private Date startDay;
    private Integer period;
    private Integer deposit;
    private String reference;
    private Integer interestRate;
    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;
}
